src/structs.rs

```diff
@@ -1,84 +1,54 @@
 use std::time::Duration;
 use std::convert::TryInto;
+use rustix::time::{Itimerspec, Timespec};
 
 use TimerState;
-use libc;
 
-// libc timespec is really awkward to work with (no traits etc)
-// so we have our own
-#[repr(C)]
-#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord)]
-struct timespec {
-    tv_sec: libc::time_t,
-    tv_nsec: libc::c_long,
-}
-
-#[repr(C)]
-#[derive(Debug, Clone, PartialEq, Eq)]
-pub struct itimerspec {
-    it_interval: timespec,
-    it_value: timespec,
-}
-
-impl itimerspec {
-    pub fn null() -> itimerspec {
-        itimerspec { it_interval: TS_NULL, it_value: TS_NULL }
-    }
-}
-
-const TS_NULL: timespec = timespec { tv_sec: 0, tv_nsec: 0 };
+const TS_NULL: Timespec = Timespec { tv_sec: 0, tv_nsec: 0 };
 
-impl From<Duration> for timespec {
-    fn from(d: Duration) -> timespec {
-        timespec {
-            tv_sec: d.as_secs() as libc::time_t,
-            // This can theoretically fail on 32-bit (`u32` to `c_long = i32`)
-            tv_nsec: d.subsec_nanos().try_into()
-                .expect("timespec overflow when converting from Duration"),
-        }
+fn to_timespec(d: Duration) -> Timespec {
+    Timespec {
+        tv_sec: d.as_secs().try_into().unwrap(),
+        tv_nsec: d.subsec_nanos() as _,
```

Comment:
What's the rationale for removing the overflow check here?

Reply:
`subsec_nanos` is [documented](https://doc.rust-lang.org/stable/std/time/struct.Duration.html#method.subsec_nanos) to always return a value less than one billion, so it'll always be representable as either `i32` or `u32` without overflow.